PyTorch nightly CI raises assertion error at installation

### Bug description

Our nightly jobs no longer succeeed in installing the PyTorch nightlies.

https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=173432&view=logs&j=5b0799f7-725e-5b16-9b83-c0a5a25d03f0&t=8cd2c85f-5065-5403-367b-21d4d1955196




### What version are you seeing the problem on?

master

### How to reproduce the bug

```python
See CI output in recent PRs
```


### Error messages and logs

```

Found existing installation: torch 2.0.1+cu118
Uninstalling torch-2.0.1+cu118:
  Successfully uninstalled torch-2.0.1+cu118
Found existing installation: torchvision 0.15.2+cu118
Uninstalling torchvision-0.15.2+cu118:
  Successfully uninstalled torchvision-0.15.2+cu118
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Looking in indexes: https://download.pytorch.org/whl/nightly/cu118
Collecting torch
  Downloading https://download.pytorch.org/whl/nightly/cu118/torch-2.2.0.dev20230906%2Bcu118-cp310-cp310-linux_x86_64.whl (2337.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 GB 45.4 MB/s eta 0:00:00
Collecting torchvision
  Downloading https://download.pytorch.org/whl/nightly/cu118/torchvision-0.16.0.dev20230906%2Bcu118-cp310-cp310-linux_x86_64.whl (6.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.2/6.2 MB 49.1 MB/s eta 0:00:00
Requirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from torch) (3.12.3)
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.10/dist-packages (from torch) (4.7.1)
Requirement already satisfied: sympy in /usr/local/lib/python3.10/dist-packages (from torch) (1.12)
Requirement already satisfied: networkx in /usr/local/lib/python3.10/dist-packages (from torch) (3.1)
Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/dist-packages (from torch) (3.1.2)
Requirement already satisfied: fsspec in /usr/local/lib/python3.10/dist-packages (from torch) (2023.6.0)
Collecting pytorch-triton==2.1.0+e6216047b8 (from torch)
  Downloading https://download.pytorch.org/whl/nightly/pytorch_triton-2.1.0%2Be6216047b8-cp310-cp310-linux_x86_64.whl (96.4 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.4/96.4 MB 46.5 MB/s eta 0:00:00
Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from torchvision) (1.25.2)
Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from torchvision) (2.31.0)
Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in /usr/local/lib/python3.10/dist-packages (from torchvision) (10.0.0)
Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2->torch) (2.1.3)
Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->torchvision) (3.2.0)
Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3/dist-packages (from requests->torchvision) (2.8)
Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/lib/python3/dist-packages (from requests->torchvision) (1.25.8)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3/dist-packages (from requests->torchvision) (2019.11.28)
Requirement already satisfied: mpmath>=0.19 in /usr/local/lib/python3.10/dist-packages (from sympy->torch) (1.3.0)
Installing collected packages: pytorch-triton, torch, torchvision
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
lightning 2.1.0rc0 requires torch<2.1.0,>=1.11.0, but you have torch 2.2.0.dev20230906+cu118 which is incompatible.
Successfully installed pytorch-triton-2.1.0+e6216047b8 torch-2.2.0.dev20230906+cu118 torchvision-0.16.0.dev20230906+cu118
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AssertionError: 2.2.0.dev20230906+cu118
##[error]Bash exited with code '1'.
Finishing: Bump to nightly

```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>


### More info

_No response_

cc @carmocca @borda

--- a/.azure/gpu-tests-fabric.yml
+++ b/.azure/gpu-tests-fabric.yml
@@ -66,7 +66,7 @@ jobs:
           image: "pytorchlightning/pytorch_lightning:base-cuda-py3.10-torch2.0-cuda11.8.0"
           IS_NIGHTLY: "false"
           PACKAGE_NAME: "lightning"
-        'Lightning | nightly':
+        'Lightning | RC':
           image: "pytorchlightning/pytorch_lightning:base-cuda-py3.10-torch2.0-cuda11.8.0"
           IS_NIGHTLY: "true"
           PACKAGE_NAME: "lightning"
@@ -115,11 +115,11 @@ jobs:
 
     - bash: |
         pip uninstall -y torch torchvision
-        pip install torch torchvision -U --pre --no-cache --index-url https://download.pytorch.org/whl/nightly/cu${CUDA_VERSION_MM%}
+        pip install torch torchvision -U --pre --no-cache --index-url https://download.pytorch.org/whl/test/cu${CUDA_VERSION_MM%}
         python -c "from torch import __version__ as ver; assert ver.startswith('2.1.0'), ver"
       # without succeeded this could run even if the job has already failed
       condition: and(succeeded(), eq(variables.IS_NIGHTLY, 'true'))
-      displayName: 'Bump to nightly'
+      displayName: 'Bump to RC'
 
     - bash: |
         set -e

--- a/.azure/gpu-tests-pytorch.yml
+++ b/.azure/gpu-tests-pytorch.yml
@@ -58,7 +58,7 @@ jobs:
           image: "pytorchlightning/pytorch_lightning:base-cuda-py3.10-torch2.0-cuda11.8.0"
           IS_NIGHTLY: "false"
           PACKAGE_NAME: "lightning"
-        'Lightning | nightly':
+        'Lightning | RC':
           image: "pytorchlightning/pytorch_lightning:base-cuda-py3.10-torch2.0-cuda11.8.0"
           IS_NIGHTLY: "true"
           PACKAGE_NAME: "lightning"
@@ -125,11 +125,11 @@ jobs:
 
     - bash: |
         pip uninstall -y torch torchvision
-        pip install torch torchvision -U --pre --no-cache --index-url https://download.pytorch.org/whl/nightly/cu${CUDA_VERSION_MM%}
+        pip install torch torchvision -U --pre --no-cache --index-url https://download.pytorch.org/whl/test/cu${CUDA_VERSION_MM%}
         python -c "from torch import __version__ as ver; assert ver.startswith('2.1.0'), ver"
       # without succeeded this could run even if the job has already failed
       condition: and(succeeded(), eq(variables.IS_NIGHTLY, 'true'))
-      displayName: 'Bump to nightly'
+      displayName: 'Bump to RC'
 
     - bash: pip uninstall -y lightning
       # without succeeded this could run even if the job has already failed

